apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestColdStartLatencyActivity.java

```diff
@@ -0,0 +1,209 @@
+        mInputButton.setEnabled(!running);
+        mLowLatencyCheckBox.setEnabled(!running);
+        mMmapCheckBox.setEnabled(!running);
+        mExclusiveCheckBox.setEnabled(!running);
+        mStartStabilizeDelaySpinner.setEnabled(!running);
+        mCloseOpenDelaySpinner.setEnabled(!running);
+        mOpenStartDelaySpinner.setEnabled(!running);
+    }
+
+    protected class MyStreamSniffer extends Thread {
+        boolean enabled = true;
+        StringBuffer statusBuffer = new StringBuffer();
+        int loopCount;
+
+        @Override
+        public void run() {
+            boolean useInput = mInputButton.isChecked();
+            boolean useLowLatency = mLowLatencyCheckBox.isChecked();
+            boolean useMmap = mMmapCheckBox.isChecked();
+            boolean useExclusive = mExclusiveCheckBox.isChecked();
+            Log.d(TAG,(useInput ? "IN" : "OUT")
+                    + ", " + (useLowLatency ? "LOW_LATENCY" : "NOT LOW_LATENCY")
+                    + ", " + (useMmap ? "MMAP" : "NOT MMAP")
+                    + ", " + (useExclusive ? "EXCLUSIVE" : "SHARED"));
+            String closeSleepTimeText =
+                    (String) mCloseOpenDelaySpinner.getAdapter().getItem(
+                            mCloseOpenDelaySpinner.getSelectedItemPosition());
+            int closedSleepTimeMillis = Integer.parseInt(closeSleepTimeText);
+            Log.d(TAG, "Sleep before open time = " + closedSleepTimeMillis + " msec");
+            String openSleepTimeText = (String) mOpenStartDelaySpinner.getAdapter().getItem(
+                    mOpenStartDelaySpinner.getSelectedItemPosition());
+            int openSleepTimeMillis = Integer.parseInt(openSleepTimeText);
+            Log.d(TAG, "Sleep after open Time = " + openSleepTimeMillis + " msec");
+            String startStabilizeTimeText = (String) mStartStabilizeDelaySpinner.getAdapter().getItem(
+                    mStartStabilizeDelaySpinner.getSelectedItemPosition());
+            int startSleepTimeMillis = Integer.parseInt(startStabilizeTimeText);
+            Log.d(TAG, "Sleep after start Time = " + startSleepTimeMillis + " msec");
+            while (enabled) {
+                loopCount++;
+                try {
+                    sleep(closedSleepTimeMillis);
+                    openStream(useInput, useLowLatency, useMmap, useExclusive);
+                    log("-------#" + loopCount + " Device Id: " + getDeviceId());
+                    log("open() Latency: " + getOpenTimeMicros() / 1000 + " msec");
+                    sleep(openSleepTimeMillis);
+                    startStream();
+                    log("requestStart() Latency: " + getStartTimeMicros() / 1000 + " msec");
+                    sleep(startSleepTimeMillis);
+                    log("Cold Start Latency: " + getColdStartTimeMicros() / 1000 + " msec");
+                    stopStream();
+                } catch (InterruptedException e) {
+                } finally {
+                    stopStream();
```

Comment:
Should also call closeStream(). The openStream() may call close() but we want the hardware to rest
while calling sleep(closedSleepTimeMillis).